src/toil/cwl/cwltoil.py

```diff
@@ -1754,17 +2524,59 @@ def filtered_secondary_files(unfiltered_secondary_files: dict) -> list:
         if ("$(" not in sf_bn) and ("${" not in sf_bn):
             if ("$(" not in sf_loc) and ("${" not in sf_loc):
                 intermediate_secondary_files.append(sf)
-    # remove secondary files that are not present in the filestore
-    # i.e. 'file://' only gets converted to 'toilfs:' upon a successful import
+    # remove secondary files that are not present in the filestore or pointing
+    # to existant things on disk
     for sf in intermediate_secondary_files:
         sf_loc = sf.get("location", "")
-        # directories aren't imported, so don't worry about them
-        if sf_loc.startswith("toilfs:") or sf.get("class", "") == "Directory":
+        if (sf_loc.startswith("toilfs:") or
+            sf_loc.startswith("toildir:") or
+            sf_loc.startswith("_:") or
+            sf.get("class", "") == "Directory"):
+            # Pass imported files, and all Directories
+            final_secondary_files.append(sf)
+        elif (sf_loc.startswith("file:") and
+              os.path.exists(schema_salad.ref_resolver.uri_file_path(sf_loc))):
+            # Pass things that exist on disk (which we presumably declined to
+            # import because we aren't using the file store)
             final_secondary_files.append(sf)
     return final_secondary_files
 
+def scan_for_unsupported_requirements(tool: Process, bypass_file_store: bool = False) -> None:
+    """
+    Scan the given CWL tool for any unsupported optional features.
+
+    If it has them, raise an informative UnsupportedRequirement.
+
+    :param tool: The CWL tool to check for unsupported requirements.
+
+    :param bypass_file_store: True if the Toil file store is not being used to
+    transport files between nodes, and raw origin node file:// URIs are exposed
+    to tools instead.
 
-def determine_load_listing(tool: ToilCommandLineTool):
+    """
+
+    # First we make a list of everything we can't support
+    unsupported = []
+
+    if not bypass_file_store:
+        # The Toil file store can't support in-place update
+        unsupported.append("InplaceUpdateRequirement")
+
+    # TODO: Can we change this to a set of supported requirements instead?
+    # Otherwise when new ones show up we assume we have them even when we
+    # don't.
+
+    for req_name in unsupported:
+        # Grab each reqirement that might be a problem
+        req, is_mandatory = tool.get_requirement(req_name)
+        if req and is_mandatory:
+            # The tool actualy uses this one, and it isn't just a hint.
+            # Complain.
+            raise UnsupportedRequirement(
+                f"Toil cannot support {req_name} in the current configuration"
```

Comment:
Since we only have one unsupported feature, we should add a special message to the user of which option to use to enable `InplaceUpdateRequirement` and that is must be a shared file system.